Allow to create attribute values from product form

elwoodxblues opened the pull request:
This PR is follow-up to #1964. Just as with variant attributes, it introduces the ability to create new attribute values from the product form. 

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

akjanik commented on saleor/dashboard/product/forms.py:
```diff
@@ -209,10 +209,16 @@ def save(self, commit=True):
+
+            # if the passed attribute value is a string,
+            # create the attribute value.
+            if not isinstance(value, AttributeChoiceValue):
```
This block is the same as in `VariantAttributeForm`, maybe we can make a function from it?

elwoodxblues commented:
@akjanik You were right about duplicated code, so I've extracted a form mixin that handles dynamic attributes fields in the same way in both product and variant form. I think the code looks a bit better now.

Also, I've taken the opportunity and dropped `get_attribute` and `set_attribute` functions from models, that were used mostly in tests. There is definitely too many confusing utility functions related to attributes, that we should eventually organize.

mad-anne commented on saleor/dashboard/product/forms.py:
```diff
@@ -137,7 +137,53 @@ def clean(self):
+
+    def iter_attribute_fields(self):
+        for attr in self.available_attributes:
+            yield self[attr.get_formfield_name()]
```
Shouldn't it be rather `self[attr.get_formfield_name()]`?

elwoodxblues replied:
You can access fields in this way as well. It was here before so I didn't change it.
